From e7b8c42c6bf02f4c2e24b015a12cd9edad094644 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Sun, 14 Nov 2021 12:36:52 -0600
Subject: [PATCH 046/117] dt-bindings: hwlock: sun6i: Add per-SoC compatibles

While all implementations of this hardware appear to be indentical, it
is possible that some difference exists. To be safe, add a compatible
for each SoC integration, using the A31 compatible only as a fallback.

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 .../hwlock/allwinner,sun6i-a31-hwspinlock.yaml   | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

--- a/Documentation/devicetree/bindings/hwlock/allwinner,sun6i-a31-hwspinlock.yaml
+++ b/Documentation/devicetree/bindings/hwlock/allwinner,sun6i-a31-hwspinlock.yaml
@@ -15,7 +15,21 @@ description:
 
 properties:
   compatible:
-    const: allwinner,sun6i-a31-hwspinlock
+    oneOf:
+      - items:
+          - enum:
+              - allwinner,sun8i-a23-hwspinlock
+              - allwinner,sun8i-a33-hwspinlock
+              - allwinner,sun8i-a50-hwspinlock
+              - allwinner,sun8i-a83t-hwspinlock
+              - allwinner,sun8i-h3-hwspinlock
+              - allwinner,sun9i-a80-hwspinlock
+              - allwinner,sun20i-d1-hwspinlock
+              - allwinner,sun50i-a64-hwspinlock
+              - allwinner,sun50i-h6-hwspinlock
+              - allwinner,sun50i-r329-hwspinlock
+          - const: allwinner,sun6i-a31-hwspinlock
+      - const: allwinner,sun6i-a31-hwspinlock
 
   reg:
     maxItems: 1
